docker-entrypoint.sh

```diff
@@ -86,8 +86,9 @@ fi
 # Boot the VM.
 #
 
-mkdir -p "$raw_cloud_config_dir"
-cat "$CLOUD_CONFIG_PATH" | base64 -d | gunzip > "$raw_cloud_config_path"
+mkdir -p "${RAW_CLOUD_CONFIG_DIR}"
+cat "${CLOUD_CONFIG_PATH}" | base64 -d | gunzip > "${RAW_CLOUD_CONFIG_PATH}"
+echo "hostname: '${HOSTNAME}'" >> "${RAW_CLOUD_CONFIG_PATH}"
```

Comment:
This is the real change. I noticed that we do not define the hostname in the cloud config templates at all. So there is nothing we can search/replace like suggested in the original issue. I would just go with this dirty trick since the other way around wouldn't be nicer anyway. This should cause the proper hostname being set in the resulting cloud config yaml.